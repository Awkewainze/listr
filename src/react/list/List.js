import React from 'react';
import './List.scss';
import TransitionGroup from 'react-transition-group/cjs/TransitionGroup';

/*
 * This component will throw a warning during deletion of items.
 * See https://github.com/reactjs/react-transition-group/issues/429
 * and https://github.com/reactjs/react-transition-group/issues/606
 * We need to wait for react-transition-group to come to a solution to resolve this.
 */
const List = ({ children }) => (
  <TransitionGroup className="ActiveList">
    {children}
  </TransitionGroup>
);

export default List;