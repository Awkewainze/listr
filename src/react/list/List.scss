@import '../../variables';
@import '../../mixins';

.List {
  grid-column: 1 / 5;

  .ListItem {

    @include transitionState($willEnter, $exiting) {
      opacity: 0;
    }

    @include transitionState($exiting) {
      transition: opacity 200ms ease;
    }

    @include transitionState($entered) {
      transition: opacity 200ms ease;
      opacity: 1;
    }
  }
}