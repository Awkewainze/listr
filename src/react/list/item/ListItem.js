import React from 'react';
import './ListItem.scss';
import Checkbox from './Checkbox';
import DeleteButton from './DeleteButton';

const ListItem = ({
  item,
  itemId,
  index,
  checked = false,
  onCheck,
  onDelete
}) => (
  <div className="ListItem">
    <Checkbox className="ListItem__checkbox" checked={checked} index={index} itemId={itemId} onChange={onCheck} />
    <DeleteButton itemId={itemId} onDelete={onDelete} />
    <p className={`ListItem__text ${checked ? 'ListItem__text--checked' : ''}`}>
      {item}
    </p>
  </div>
);

export default ListItem;