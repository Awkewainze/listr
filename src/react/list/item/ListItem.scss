@import '../../../variables';
@import '../../../mixins';

.ListItem {
  display: flex;
  border-bottom: 1px solid #afafaf;
  margin: 6px 0;
  justify-content: flex-start;
  padding: 0 20px;

  @include minWidth($size-mobile) {
    .DeleteButton {
      opacity: 0;
    }

    &:hover {
      border-color: #7f7f7f;

      .DeleteButton {
        opacity: 1;
      }
    }
  }

  .ListItem__text {
    padding: 20px 0;
  }

  .ListItem__checkbox {
    cursor: pointer;
    margin-right: 12px;
  }

  .ListItem__text {
    flex: 1;
    overflow: scroll;
    margin: 12px 0;
    line-height: 26px;
  }

  .ListItem__text--checked {
    text-decoration: line-through;
    opacity: 0.75;
  }

  @include minWidth($size-mobile) {
    .ListItem__checkbox {
      min-width: 72px;
      margin-right: 0;
    }
  }
}