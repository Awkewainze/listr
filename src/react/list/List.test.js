import React from 'react';
import { shallow } from 'enzyme';
import List from './List';
import ListItem from './item/ListItem';

const testItems = [
  {
    item: 'apples',
    checked: false,
    id: '123123'
  },
  {
    item: 'oranges',
    checked: true,
    id: '321321'
  }
];

const TEST_PROPS = {
  items: testItems
};

const render = ({ items }) => <List>{items.map(renderItem)}</List>;

const renderItem = (itemProps, index) => <ListItem key={index} {...itemProps} />;

it('matches snapshot', () => {
  expect(shallow(render(TEST_PROPS))).toMatchSnapshot();
});