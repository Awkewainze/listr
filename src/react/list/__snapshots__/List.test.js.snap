// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<TransitionGroup
  childFactory={[Function]}
  className="ActiveList"
  component="div"
>
  <ListItem
    checked={false}
    id="123123"
    item="apples"
    key="0"
  />
  <ListItem
    checked={true}
    id="321321"
    item="oranges"
    key="1"
  />
</TransitionGroup>
`;
